File: backend/src/main/java/com/ares/backend/entity/PasoReceta.java
package com.ares.backend.entity;

import jakarta.persistence.*;
import lombok.*;

/**
 * Entidad que representa un paso individual de una receta.
 * Los pasos están ordenados secuencialmente para guiar la elaboración.
 *
 * @author Ares
 * @version 1.0
 */
@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@ToString
@Table(name = "pasos_receta")
public class PasoReceta {

    /**
     * Identificador único del paso.
     */
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    /**
     * Orden del paso en la secuencia de la receta (1, 2, 3...).
     */
    @Column(nullable = false)
    private Integer orden;

    /**
     * Descripción detallada del paso a realizar.
     */
    @Column(nullable = false, length = 500)
    private String descripcion;

    /**
     * Receta a la que pertenece este paso.
     */
    @ManyToOne
    @JoinColumn(name = "receta_id", nullable = false)
    private Receta receta;

    /**
     * Constructor con parámetros para crear un paso de receta.
     *
     * @param orden Orden del paso en la secuencia
     * @param descripcion Descripción del paso
     * @param receta Receta a la que pertenece
     */
    public PasoReceta(Integer orden, String descripcion, Receta receta) {
        this.orden = orden;
        this.descripcion = descripcion;
        this.receta = receta;
    }
}

File: frontend/flutter_app/lib/models/paso_receta.dart
/// Un paso individual dentro de una receta.
///
/// [orden] determina la secuencia de visualización (1-based).
class PasoReceta {
  final int    id;
  final int    orden;
  final String descripcion;
  final int    recetaId;

  const PasoReceta({
    required this.id,
    required this.orden,
    required this.descripcion,
    required this.recetaId,
  });

  factory PasoReceta.fromJson(Map<String, dynamic> json) => PasoReceta(
        id:          json['id']          as int,
        orden:       json['orden']       as int,
        descripcion: json['descripcion'] as String,
        recetaId:    json['recetaId']    as int,
      );

  Map<String, dynamic> toJson() => {
        'id':          id,
        'orden':       orden,
        'descripcion': descripcion,
        'recetaId':    recetaId,
      };

  PasoReceta copyWith({
    int?    id,
    int?    orden,
    String? descripcion,
    int?    recetaId,
  }) =>
      PasoReceta(
        id:          id          ?? this.id,
        orden:       orden       ?? this.orden,
        descripcion: descripcion ?? this.descripcion,
        recetaId:    recetaId    ?? this.recetaId,
      );
}